### PR TITLE
feat(emit-on-window-close): enable queries with `EMIT ON WINDOW CLOSE` keyword

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR

- Allows `LogicalXxx::to_stream` to be aware of `EMIT ON WINDOW CLOSE` mode by adding a field to `ToStreamContext`. (Later in #9597 we will have our first operator that generates EOWC version of stream node for EOWC emit mode.)
- Adds `emit_on_window_close` field to `PlanBase` so that every plan node has to decide whether it's EOWC.
- Enforces EOWC property by adding a `StreamSort` above the generated stream plan root when user required EOWC mode.
- Supports EOWC mode in planner test.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- SQL commands, functions, and operators

### Release note

Support `EMIT ON WINDOW CLOSE` mode for queries on sources with watermark column.

Example:

```sql
CREATE SOURCE t (a TIMESTAMP, b INT, WATERMARK FOR a AS a - INTERVAL '5 minutes') WITH ...;
CREATE MATERIALIZED VIEW mv EMIT ON WINDOW CLOSE AS
SELECT
    window_start, max(b)
FROM tumble(t, a, INTERVAL '1 hour')
GROUP BY window_start;
```

### CODE REVIEW

**Review by soundOfDestiny — APPROVED** [review]
LGTM

File: src/frontend/src/handler/create_mv.rs [hunk 1]
```diff
@@ -79,6 +79,7 @@ pub fn gen_create_mv_plan(
     query: Query,
     name: ObjectName,
     columns: Vec<Ident>,
+    emit_mode: Option<EmitMode>,
```
Comment (st1page) [inline src/frontend/src/handler/create_mv.rs:82]: qmaybe adding it on the planRoot is better? It is more like a "query's property". I think we will support the EOWC sink soon.

Comment (richardchien) [reply src/frontend/src/handler/create_mv.rs:82]: Looks like it's more like `WITH`. Maybe we can refactor it later when needed😇

**Review by st1page — APPROVED** [review]
LGTM

**Pull request merged**